src/main/java/net/logstash/logback/layout/CompositeJsonLayout.java

```diff
@@ -82,26 +78,29 @@ public String doLayout(Event event) {
             throw new IllegalStateException("Layout is not started");
         }
         
-        ReusableByteBuffer buffer = this.bufferPool.acquire();
-        try (OutputStreamWriter writer = new OutputStreamWriter(buffer)) {
+        try (ReusableJsonFormatterPool<Event>.ReusableJsonFormatter cachedFormatter = formatterPool.acquire()) {
+            writeEvent(cachedFormatter, event);
+            return new String(cachedFormatter.getBuffer().toByteArray());
+        
+        } catch (IOException e) {
+            addWarn("Error formatting logging event", e);
+            return null;
+            
+        }
+    }
+
+    private void writeEvent(ReusableJsonFormatterPool<Event>.ReusableJsonFormatter cachedFormatter, Event event) throws IOException {
+        try (Writer writer = new OutputStreamWriter(cachedFormatter.getBuffer())) {
```

Comment:
Creating a new OutputStreamWriter every time is not very efficient... but was already the case before these changes. Layout support will be deprecated and removed in the next major release... therefore I decided not to spend too much time and complexity in a more efficient implementation.